fix PIE generation for MIPS targets

http://sourceware.org/bugzilla/show_bug.cgi?id=10858
http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=526961
http://bugs.gentoo.org/301834

===================================================================
RCS file: /cvs/src/src/bfd/elfxx-mips.c,v
retrieving revision 1.266
retrieving revision 1.267
diff -u -r1.266 -r1.267
--- src/bfd/elfxx-mips.c	2010/02/04 09:16:41	1.266
+++ src/bfd/elfxx-mips.c	2010/02/24 14:21:51	1.267
@@ -5755,9 +5755,7 @@
 
   /* We must now calculate the dynamic symbol table index to use
      in the relocation.  */
-  if (h != NULL
-      && (!h->root.def_regular
-	  || (info->shared && !info->symbolic && !h->root.forced_local)))
+  if (h != NULL && ! SYMBOL_REFERENCES_LOCAL (info, &h->root))
     {
       indx = h->root.dynindx;
       if (SGI_COMPAT (output_bfd))
